### ISSUE TITLE
Remove GridSet class from Public API

### ISSUE THREAD

**Issue by VeckoTheGecko** [OceanParcels/Parcels#1698]
I don't think the `GridSet` object really needs to be called by users at all, as they are not manually defining gridsets. All the grid information is available to the users via the `Field.grid` attributes, and `FieldSet.gridset` attributes. 

I also think that the `GridSet` itself can be majorly simplified (ie, `add_grid()` removed/majorly refactored, grids being an actual `set`, turning it into a dataclass) to the point that we could maybe even outright replace it with some utility functions/processors that pull grid information from the Fields. That's out of topic for this issue, just a hint to where I can see this going.

@erikvansebille  any insight onto the history of the gridset class?

**Comment by erikvansebille** [comment]
Yes, good point. The `GridSet` is perhaps a bit over-engineered. The idea was to mimic the `ParticleSet` architecture/structure, but the `GridSet` is, as you say, much simpler; essentially just a list (set?) of `Grids`. 

So yes, there could certainly be some cleaning here!
